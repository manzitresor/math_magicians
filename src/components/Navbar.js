import React from 'react';
import { Link } from 'react-router-dom';

const links = [
  { path: '/', name: 'Home' },
  { path: 'calculator', name: 'Calculator' },
  { path: '/quote', name: 'Quote' },
];

const logoStyles = {
  textDecoration: 'none',
  color: 'black',
  fontSize: 25,
  letterSpacing: 5,
  marginLeft: 20,
};

function Navbar() {
  return (
    <>
      <div className="navbar">
        <Link to="/" style={logoStyles}>Math Magicians</Link>
        <ul className="nav-items">
          {
        links.map((link) => (
          <li key={link.name}>
            <Link to={link.path} style={{ textDecoration: 'none', color: 'black' }}>{link.name}</Link>
          </li>
        ))
    }
        </ul>
      </div>
    </>
  );
}

export default Navbar;
