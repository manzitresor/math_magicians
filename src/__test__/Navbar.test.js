import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Navbar from '../components/Navbar';

describe('NavBar component', () => {
  test('renders correctly', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    const tree = renderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    renderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    ).toJSON();
    const navbars = screen.getAllByRole('link');
    expect(navbars).toHaveLength(4);
  });
});
