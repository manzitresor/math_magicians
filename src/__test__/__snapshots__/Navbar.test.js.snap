// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders correctly 1`] = `
<div
  className="navbar"
>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "rgb(220, 145, 75)",
        "fontSize": 25,
        "fontWeight": "bold",
        "letterSpacing": 5,
        "marginLeft": 20,
        "textDecoration": "none",
      }
    }
  >
    Math Magicians
  </a>
  <ul
    className="nav-items"
  >
    <li>
      <a
        className="nav-item"
        href="/"
        onClick={[Function]}
      >
        Home
         
        |
      </a>
    </li>
    <li>
      <a
        className="nav-item"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
         
        |
      </a>
    </li>
    <li>
      <a
        className="nav-item"
        href="/quote"
        onClick={[Function]}
      >
        Quote
         
        |
      </a>
    </li>
  </ul>
</div>
`;
